### PR TITLE
[RSDK-6720] fix replay movement sensors

### PR DESCRIPTION
**Pull request by JohnN193**
https://viam.atlassian.net/browse/RSDK-6720

with the tabular data capture changes, replay movement sensors could receive data from app, but would fail to deserialize the message from proto. This resulted in replay movement sensors returning a value of 0 for everything, which we only noticed when testing with odometry

### CODE REVIEW

File: components/movementsensor/replay/replay.go [hunk 1]
```diff
@@ -214,11 +217,18 @@ func (replay *replayMovementSensor) Position(ctx context.Context, extra map[stri
...
+	altitude, ok := data.GetFields()["altitude_m"]
+	if !ok {
+		return nil, 0, errBadData
+	}
 	return geo.NewPoint(
-			coordStruct.GetFields()["latitude"].GetNumberValue(),
-			coordStruct.GetFields()["longitude"].GetNumberValue()),
-		data.GetFields()["altitude_m"].GetNumberValue(), nil
+			coordStruct.GetStructValue().GetFields()["latitude"].GetNumberValue(),
+			coordStruct.GetStructValue().GetFields()["longitude"].GetNumberValue()),
```
Comment (JohnN193) [inline components/movementsensor/replay/replay.go:230]: did not test if the subfields matched. felt like overkill for this PR but I can if ya want

File: components/movementsensor/replay/replay_utils_test.go [hunk 2]
```diff
@@ -241,6 +247,69 @@ func createDataByMovementSensorMethod(method method, index int) *structpb.Struct
...
+			"theta": structpb.NewNumberValue(orientationData[index].Theta),
+		}
+		data.Fields = map[string]*structpb.Value{
+			"orientation": structpb.NewStructValue(&orient),
+		}
+	}
+	return &data
+}
+
+// createBadDataByMovementSensorMethod will create the mocked structpb.Struct containing the next data returned by calls in tabular data.
```
Comment (vijayvuyyuru) [inline components/movementsensor/replay/replay_utils_test.go:299]: I think this comment needs to be updated

**Review by vijayvuyyuru — APPROVED** [review]
Look good to me! Nice work on expanding the test coverage and sorry for not catching this the first time around. One comment needs updating but otherwise 🚀

File: components/movementsensor/replay/replay.go (hunk 1 above)
Comment (vijayvuyyuru) [reply components/movementsensor/replay/replay.go:230]: I think since its data from us, being used by us its fine to leave as is.

**Pull request merged**